### PR TITLE
doc/stdenv: put note about remote builds with breakpointHook into a <note>

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

With remote builds, the sandbox can't be accessed by `cntr` as it is on
a different machine. I decided to document this in the manual as it took
me admittedly too much time to figure this out.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: doc/stdenv.xml [hunk 1]
```diff
@@ -2657,6 +2657,14 @@ addEnvHooks "$hostOffset" myBashFunction
        When a build failure happens there will be an instruction printed that
        shows how to attach with <literal>cntr</literal> to the build sandbox.
       </para>
+      <para>
+       <emphasis>Note:</emphasis> this won't work with remote builds as the
```
Comment (worldofpeace) [inline doc/stdenv.xml]: Why not use `<note></note>`?

Comment (Ma27) [reply doc/stdenv.xml]: Admittedly I didn't know about this, thanks!

**Comment by Ma27** [comment]
@worldofpeace as this applies to `release-19.03` as well I suggest we backport this, so this information is also available on the website's wiki :)

**Comment by worldofpeace** [comment]
> @worldofpeace as this applies to `release-19.03` as well I suggest we backport this, so this information is also available on the website's wiki :)

Yeah, I think typically docs changes are good to backport if possible.

File: doc/stdenv.xml [hunk 2]
```diff
@@ -2657,6 +2657,16 @@ addEnvHooks "$hostOffset" myBashFunction
        When a build failure happens there will be an instruction printed that
        shows how to attach with <literal>cntr</literal> to the build sandbox.
       </para>
+      <note>
+       <title>Caution with remote builds</title>
+       <para>
+        This won't work with remote builds as the build environment is on
+        a different machine and can't be accessed by <command>cntr</command> then.
```
Comment (worldofpeace) [inline doc/stdenv.xml]:
```suggestion
        a different machine and can't be accessed by <command>cntr</command>.
```

File: doc/stdenv.xml [hunk 3]
```diff
@@ -2657,6 +2657,16 @@ addEnvHooks "$hostOffset" myBashFunction
...
+        Remote builds can be turned off by setting <literal>--option builders ''</literal>
+        for <command>nix-build</command> or <literal>--builders ''</literal> for
```
Comment (worldofpeace) [inline doc/stdenv.xml]:
```suggestion
        or <literal>--builders ''</literal> for
```

File: doc/stdenv.xml (hunk 2 above)
Comment (Ma27) [reply doc/stdenv.xml]: @worldofpeace fixed.

**Comment by worldofpeace** [comment]
I've rendered this and noticed there's a sentence that effectively says this but too subtle
Screenshot from 2019-05-07 15 29 05

I understood this with more clarity with your note
Screenshot from 2019-05-07 15 29 41


Perhaps you can remove the detail about remote builders in that sentence?

**Comment by Ma27** [comment]
Oh damn it, that also means that I overlooked it when I was reading the docs in the first place :sweat_smile:

**Comment by worldofpeace** [comment]
> Oh damn it, that also means that I overlooked it when I was reading the docs in the first place sweat_smile

Understandable, I almost missed it also. Plus with this we can tell how it disable it.

**Review by worldofpeace — APPROVED** [review]
LGTM, everything was addressed that could be.

**Pull request merged**